src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,100 @@
 package core.basesyntax;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
+
+    private static final String SPLIT_REGEX = "\\r?\\n";
+    private static final String SEP = ",";
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final String NEW_LINE = System.lineSeparator();
+
+    private static final class Totals {
+        private final int supply;
+        private final int buy;
+
+        Totals(int supply, int buy) {
+            this.supply = supply;
+            this.buy = buy;
+        }
+    }
+
     public void getStatistic(String fromFileName, String toFileName) {
 
+        String content = readFile(fromFileName);
+        Totals totals = parseAndAggregate(content);
+        String report = buildReport(totals);
+        writeFile(toFileName, report);
+    }
+
+    private String readFile(String fromFileName) {
+        try {
+            return Files.readString(Path.of(fromFileName));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read a file: " + fromFileName + e);
```

Violation of checklist item "Don't ignore exceptions": the catch here concatenates the exception into the message and does not preserve the original exception as the cause. The checklist recommends rethrowing like `throw new RuntimeException("Can't read data from the file " + fileName, e);` so the original exception is preserved. Change this to include the original exception as the cause and provide a clear message (for example: `throw new RuntimeException("Can't read file: " + fromFileName, e);`).